Fix CUDA Bazel build to optionally include gmock after #104255

huydhn opened the pull request:
This reverts commit 39868b0578c18ddc194deac697d0675760de5f11.  Fixes https://github.com/pytorch/pytorch/issues/104279.

The change came from an internal codemod diff that we don't want to revert.  AFAIK, this addition is not needed as gmock has already been included https://github.com/google/googletest/blob/main/BUILD.bazel

### Testing

* OSS CUDA Bazel build should be back after this revert
* Import as D47077813 to make sure that nothing breaks internally

malfet commented on c10/cuda/test/build.bzl:
```diff
@@ -17,9 +17,11 @@ def define_targets(rules):
         target_compatible_with = rules.requires_cuda_enabled(),
         deps = [
             "//c10/cuda",
-            "@com_google_googletest//:gmock",
```
Instead of doing that, just add `gtest_deps` parameter to `define_targets` and pass one deps from TARGETS file and another from BUILD.bazel

huydhn committed:
Fix CUDA Bazel build to optionally include gmock after #104255 (#104308)

Summary:
This reverts commit 39868b0578c18ddc194deac697d0675760de5f11.  Fixes https://github.com/pytorch/pytorch/issues/104279.

The change came from an internal codemod diff that we don't want to revert.  AFAIK, this addition is not needed as gmock has already been included https://github.com/google/googletest/blob/main/BUILD.bazel

### Testing

* OSS CUDA Bazel build should be back after this revert
* Import as D47077813 to make sure that nothing breaks internally

Pull Request resolved: https://github.com/pytorch/pytorch/pull/104308

Test Plan: buck build //caffe2/c10/cuda/test

Reviewed By: malfet, kit1980

Differential Revision: D47077813

Pulled By: huydhn

fbshipit-source-id: 254a7b196c153ec3927a03de10bb7ace58f2367a

huydhn commented:
@pytorchbot merge
